src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs

```diff
@@ -1003,7 +1011,7 @@ protected override void EndProcessing()
 
                     BuildPrivateDataInModuleManifest(result, streamWriter);
 
-                    BuildModuleManifest(result, "HelpInfoURI", Modules.HelpInfoURI, !string.IsNullOrEmpty(_helpInfoUri), () => QuoteName(_helpInfoUri), streamWriter);
+                    BuildModuleManifest(result, "HelpInfoURI", Modules.HelpInfoURI, !string.IsNullOrEmpty(_helpInfoUri), () => QuoteName((null != _helpInfoUri) ? new Uri(_helpInfoUri) : null), streamWriter);
```

Comment:
This statement is getting rather convoluted.  Suggest breaking out the QuoteName logic for readability.